common/src/main/java/com/wynntils/handlers/item/ItemHandler.java

```diff
@@ -251,6 +260,33 @@ private ItemAnnotation calculateAnnotation(ItemStack itemStack, StyledText name)
         return annotation;
     }
 
+    private StyledText simplifyName(StyledText name) {
+        String full = name.getString();
+
+        for (Pattern p : simplifiablePatterns) {
+            Matcher m = p.matcher(full);
```

Comment:
The problem you are trying to solve here (to copy the surrounding formatting to a match group of a StyledText), should be solved in StyledText itself. My first reaction was "we already do this" but then I realize this is probably only done for StyledText.split(). 

@kristofbolyai is the main architect of StyledText, so I'd like for him to weigh in. My personal, initial idea is that we should have something like `StyledText.formattedSubstring()` that you could use like:
```
if (m.matches()) {
    return name.formattedSubstring(m.group(1));
}
```

and it will basically do what you do, but properly. (Your code will fail e.g. if there are consecutive formatting codes, like a color and then bold.), presumably it will look something like this pseudo-code:
```
public StyledText formattedSubstring(String substring) {
  StyledText[] parts = this.split(substring);
  if (parts.length < 2) return StyledText.EMPTY;
  StyledTextFormatting format = parts[1];
  return new StyledText(substring, format);
}
```

Reply:
Doesn't StyledText#substring already apply colors if asked? But only indexes on non-formatted text?

Reply:
Then it can perhaps be used. But we still need a (convenience) method in StyledText that takes a snippet String (like from a match group) of the string, and returns it as a StyledText with the formatting the snippet would have had in the original StyledText. 

That functionality is still needed for this to work properly, and the place to implement that logic is inside StyledText, not here.

Reply:
I am not sure if there is a proper implementation for that..?

Reply:
You mean that we do not have that functionality yet, or if it is impossible to do? I am pretty sure the former is true and the latter is false, and I'm saying this needs to be done as part of this PR.

Reply:
I mean that I don't think there can be an implementation like that. You can apply a specific style to a part, but to apply multiple, you would need to find the correct parts, split the unformatted text, apply the formats back. Reasonable, but kind of also bad. Also, this can only reasonably work on the first match it finds, which makes it only work reasonably on unique text parts.

Reply:
What about the "splitAndKeep()" (or perhaps even better "separate()"?) method I suggested in Donkey's PR? Like split, but do not throw away the part matching the split pattern. That would be useful here. 

Also, here we know there can be only a single match, but if it was possible there could be more, the calling code would know what to do. Maybe concatenate all but the first returned parts, for instance, if we wanted to throw away everything before the matching pattern? Etc.

Reply:
Sure.

Reply:
How about a `StyledTextMatcher` that can match a string with color codes (§x) onto a `StyledText` and return `StyledText` as groups? (This would do the same as `formattedSubstring` but it could neatly handle multiple matches or groups without needing to use multiple matchers.)

Reply:
Let's merge now. I added a warning for the code.